collect_app/src/main/java/org/odk/collect/android/activities/MainMenuActivity.java

```diff
@@ -454,14 +459,22 @@ protected Dialog onCreateDialog(int id) {
 
                 AlertDialog.Builder builder = new AlertDialog.Builder(this);
                 final AlertDialog passwordDialog = builder.create();
-
                 passwordDialog.setTitle(getString(R.string.enter_admin_password));
-                final EditText input = new EditText(this);
-                input.setInputType(InputType.TYPE_TEXT_VARIATION_PASSWORD);
-                input.setTransformationMethod(PasswordTransformationMethod
-                        .getInstance());
-                passwordDialog.setView(input, 20, 10, 20, 10);
-
+                LayoutInflater inflater = this.getLayoutInflater();
+                View dialogView = inflater.inflate(R.layout.dialogbox_layout, null);
+                passwordDialog.setView(dialogView, 20, 10, 20, 10);
+                final CheckBox checkBox = (CheckBox)dialogView.findViewById(R.id.checkBox);
+                final EditText input = (EditText)dialogView.findViewById(R.id.editText);
+                checkBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+                    @Override
+                    public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
+                        if(!checkBox.isChecked()) {
+                            input.setInputType(129);
```

Comment:
Ah setInputType seems to be more readable. From what I read I think transformation method is used to customize how to display the password field.

I used it before to display 'x' instead of the default.